src/App.tsx

```diff
@@ -8,7 +7,82 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+}
+
+interface User {
+  id: number;
+  name: string;
+  email: string;
+}
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [showModal, setShowModal] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterValue, setFilterValue] = useState('all');
+  const [searchValue, setSearchValue] = useState('');
+
+  useEffect(() => {
+    // eslint-disable-next-line max-len
+    fetch('https://mate-academy.github.io/react_dynamic-list-of-todos/api/todos.json')
+      .then((response) => response.json())
+      .then((data) => setTodos(data))
+      .catch((error) => (error));
+
+    // eslint-disable-next-line max-len
+    fetch('https://mate-academy.github.io/react_dynamic-list-of-todos/api/users.json')
+      .then((response) => response.json())
+      .then((data) => setUsers(data))
+      .catch((error) => (error));
+  }, []);
+
+  const handleShowTodoModal = (todo: Todo) => {
+    setSelectedTodo(todo);
+    setShowModal(true);
+  };
+
+  const handleCloseTodoModal = () => {
+    setSelectedTodo(null);
+    setShowModal(false);
+  };
+
+  const handleFilterChange = (value: string) => {
+    setFilterValue(value);
+  };
+
+  const handleSearchChange = (value: string) => {
+    setSearchValue(value);
+  };
+
+  const handleClearSearch = () => {
+    setSearchValue('');
+  };
+
+  const filteredTodos = todos.filter((todo) => {
+    const titleMatch = todo.title.toLowerCase()
+      .includes(searchValue.toLowerCase());
+
+    if (filterValue === 'all') {
+      return titleMatch;
+    }
+
+    if (filterValue === 'active') {
+      return !todo.completed && titleMatch;
+    }
+
+    if (filterValue === 'completed') {
+      return todo.completed && titleMatch;
+    }
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others